app/controllers/repositories_controller.rb

```diff
@@ -182,4 +225,24 @@ def create_equipments
         Equipment.create(name: e, repository_id: @repository.id)
       end if params['equipments'].present?
     end
+
+    def update_share_type
```

Comment:
I'm not sure I understand. If it's private the password needs to be there for any update?